data/src/main/java/com/microsoft/azure/kusto/data/KustoOperationResult.java

```diff
@@ -0,0 +1,95 @@
+package com.microsoft.azure.kusto.data;
+
+import com.microsoft.azure.kusto.data.exceptions.KustoServiceError;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.*;
+
+public class KustoOperationResult implements Iterator<KustoResultSetTable> {
+    private final static Map<String, WellKnownDataSet> tablesKindsMap = new HashMap<String, WellKnownDataSet>(){{
+        put("QueryResult",WellKnownDataSet.PrimaryResult);
+        put("QueryProperties",WellKnownDataSet.QueryProperties);
+        put("QueryStatus",WellKnownDataSet.QueryCompletionInformation);
+    }};
+    private static final String tableNamePropertyName = "Name";
+    private static final String tableIdPropertyName = "Id";
+    private static final String tableKindPropertyName = "Kind";
+    private static final String tablesListPropertyName = "Tables";
+    private static final String dataTableFrameTypePropertyName = "DataTable";
+    private static final String frameTypePropertyName = "FrameType";
+
+    private ArrayList<KustoResultSetTable> resultTables = new ArrayList<>();
+    private final Iterator<KustoResultSetTable> it;
+
+    public KustoOperationResult(String response, boolean isV2) throws JSONException, KustoServiceError {
```

Comment:
Consider pass the version instead of isV2, so if we'll add new version we wouldn't need to change the method signature.